.github/workflows/generate-go-docs.yaml

```diff
@@ -9,70 +9,121 @@ on:
-
-      - name: Get PR details
-        id: pr-details
-        run: |
-          echo "pr_number=$(jq -r .pull_request.number <<< "$GITHUB_EVENT_PATH")" >> $GITHUB_OUTPUT
-          echo "pr_branch=$(jq -r .pull_request.head.ref <<< "$GITHUB_EVENT_PATH")" >> $GITHUB_OUTPUT
-          echo "base_commit=${{ github.event.pull_request.base.sha }}" >> $GITHUB_OUTPUT
-          echo "head_commit=${{ github.event.pull_request.head.sha }}" >> $GITHUB_OUTPUT          
+      - name: Detect changes related to current PR
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: changes
+        with:
+          filters: |
+            seth:
+              - 'seth/**/*.go'
+            wasp:
+              - 'wasp/**/*.go'
+            havoc:
+              - 'havoc/**/*.go'
+            lib:
+              - 'lib/**/*.go'
+            k8s-test-runner:
+              - 'k8s-test-runner/**/*.go'
+            framework:
+              - 'framework/**/*.go'
+            # later add tools here or, if possible, make this filter dynamic so that it's created based on the go modules in the repository
 
-      - name: Install go doc generator
+      - name: Find all go modules in the repository and filter the ones that changed
         shell: bash
+        id: go-modules
+        env:
+          FILTERS: ${{ steps.changes.outputs.changes }}        
         run: |
-          go install github.com/smartcontractkit/generate-go-function-docs@latest
+          PATH=$PATH:$(go env GOPATH)/bin
+          export PATH                  
 
-      - name: Install gopls
-        shell: bash
-        run: |
-          go install golang.org/x/tools/gopls@latest
+          # Find all go projects
+          gomods_output=$(gomods 2>&1)
 
-      - name: Create a new branch
-        run: |
-          git checkout -b ${{ steps.pr-details.outputs.pr_branch }}-docs
+          # Extract the parent directories of go.mod files
+          parent_folders=$(echo "$gomods_output" | grep 'go\.mod$' | sed 's/\/go\.mod//' | sed 's/^[ \t]*//;s/[ \t]*$//' | xargs -n 1)
+
+          # Convert parent directories into a JSON matrix
+          echo "$parent_folders" | jq -R -s 'split("\n") | map(select(length > 0)) | map({folder: .})' > all_folders.json
+
+          # Filter the directories that did not changeß
+          jq --argjson filters "$FILTERS" 'map(select(.folder as $folder | $filters | index($folder)))' all_folders.json > filtered_folders.json
+
+          echo "Filtered folder List JSON"
+          cat filtered_folders.json
+
+          rm all_folders.json
 
-      - name: Generate go docs
+      - name: Generate go docs for changed projects
         shell: bash
         env:
           OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
         run: |
-          generate-go-function-docs diff -b ${{ steps.pr-details.outputs.base_commit }} -c ${{ steps.pr-details.outputs.head_commit }} -g fake_slash
+          # Add go binary to PATH
```

Comment:
Can't we just "go install" like other tools do or download the binary with curl and run it, why do we need a path now?